File: vue/src/types/order.ts
export type OrderStatus = 'CREATED' | 'CONFIRMED' | 'SHIPPED' | 'COMPLETED' | 'CANCELLED'

export interface OrderItem {
  product_id: number
  product_name: string
  quantity: number
  price: number
  subtotal?: number
  image_url?: string
}

export interface Order {
  id: number
  order_number: string
  status: OrderStatus
  total_amount: number
  shipping_address?: string
  note?: string
  items: OrderItem[]
  confirm_token?: string
  expires_at?: string
  confirmed_at?: string
  created_at: string
  updated_at?: string
}

export interface OrderCreate {
  shipping_address: string
  note?: string
}

export interface CartItem {
  id: number
  product: ProductBrief
  quantity: number
  subtotal: number
}

export interface ProductBrief {
  id: number
  name: string
  price: number
  image_url?: string
  stock?: number
}

export interface CartSummary {
  total_items: number
  total_amount: number
}

export interface CartState {
  items: CartItem[]
  summary: CartSummary | null
}

File: app/model/order.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Enum, ForeignKey, Text, Numeric
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.model.database import Base
import enum


class OrderStatus(str, enum.Enum):
    CREATED = "CREATED"
    CONFIRMED = "CONFIRMED"
    CANCELLED = "CANCELLED"


class Order(Base):
    __tablename__ = "order"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("user.id"), nullable=False)
    status = Column(Enum(OrderStatus), default=OrderStatus.CREATED)
    total_amount = Column(Numeric(10, 2), nullable=False)
    confirm_token = Column(String(255), nullable=True, unique=True)
    shipping_address = Column(Text, nullable=True)
    note = Column(Text, nullable=True)
    confirmed_at = Column(DateTime, nullable=True)
    expires_at = Column(DateTime, nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    user = relationship("User", back_populates="orders")
    items = relationship("OrderItem", back_populates="order", cascade="all, delete-orphan")
    
    @property
    def order_number(self) -> str:
        return f"ORD{self.id:08d}"


class OrderItem(Base):
    __tablename__ = "order_item"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    order_id = Column(Integer, ForeignKey("order.id"), nullable=False)
    product_id = Column(Integer, ForeignKey("product.id"), nullable=False)
    quantity = Column(Integer, nullable=False)
    price = Column(Numeric(10, 2), nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    
    order = relationship("Order", back_populates="items")
    product = relationship("Product", back_populates="order_items")